### ISSUE TITLE
New version nspanel pro 120 

### ISSUE THREAD

**Issue by kwetnico** [seaky/nspanel_pro_tools_apk#130]
Hello Sonoff has just released the new model of the nspanel pro 120. Do you think it would be possible to hack it as root and install nspanel pro tools?

[https://itead.cc/product/sonoff-nspanel-pro-smart-home-control-panel-120-type/?srsltid=AfmBOopZtdbB3G9ngBLdLtcxLywqbq7cNcObyNCIcIc44maVvYmnkN_e](url)

**Comment by seaky** [comment]
I ordered some of them but not yet in my hands. As I saw on tiktok the board is quite the same, and can be hacked.

**Comment by bobzer** [comment]
because of you ? or should I say thanks to you .. anyway I ordered also 👍 so i hope it gonna works, and if I can help, I'll be glad to

**Comment by kwetnico** [comment]
On reddit someone flashed the new model :
mine arrives today..
[here](https://www.reddit.com/r/homeassistant/comments/1f0jut8/sonoff_nspanel_pro_120/?tl=fr)

**Comment by seaky** [comment]
**Internal board almost the same**
- Rockchip PX30 replaced by RK3326S
- Layout is a bit different luckily otg-usb still exposed (thanks Sonoff)
![IMG_4804](https://github.com/user-attachments/assets/e34f7250-ca67-4fe0-bab0-7b90b6a66038)

- Easy to "open" as the nspanel86p was
![IMG_4812](https://github.com/user-attachments/assets/392170a8-0015-4150-bead-5d9f90d51797)
![IMG_4811](https://github.com/user-attachments/assets/bae115a3-c361-48d0-9e4d-a093178e8d20)

**Pros:**
- new chip RK3326
- Longer display
- landscape mode available
- proximity sensor autocalibrated the available value is 0 or 1

**Cons:**
- The device narrower, resulting the screen being narrower by ~1cm as well (can be compensated by OS font size change feature)
- proximity raw values are not available, distance can not be configured anymore

**Comment by seaky** [comment]
next nspanel version will fully support NSP120P too. I also imple,mented the device type detection so app will handle which function is available in which type of device.

**Comment by kwetnico** [comment]
Hello great news.
 I also received the link I was able to install the root and pro tools. It works. I look forward to the new version and your optimizations. Thank you

**Comment by Wraithers** [comment]
Hey! I've got a couple on the way as well, so it's great news to hear that almost everything should work as with the standard NSPanel Pro 🙌 

By any chance can you upload a video of the screen performance, @seaky, after NSPanel Pro Tools installed/rooting? Is it by any chance better/faster with the new chip, or is that just helping it power the increased display size..? Here's hoping for the former 🤞 

And thanks in advance! 🌟

**Comment by bobzer** [comment]
Hi, I just received two of them, I'm not sure what the correct step, I'm
kind of afraid to brick them. Can we use the same tuto that for the 86
version?

Le sam. 31 août 2024, 15 h 53, Wraithers ***@***.***> a
écrit :

**Comment by kwetnico** [comment]
Hello 

I set my nspanel pro 120 so that the screen goes to sleep after 2 minutes of no use. It works but every time I wake it up with a hand gesture and it works I see that it restarts a connection to the wifi network (as if the entire device went to sleep) I have incorrectly set an option?
 Thank you

**Comment by kwetnico** [comment]
> next nspanel version will fully support NSP120P too. I also imple,mented the device type detection so app will handle which function is available in which type of device.

What do you think about releasing the new version compatible with the NSpanel pro 120, thanks ?

**Comment by kwetnico** [comment]
> How did you test it? Do you use orignal app or HA?
> 
> Mine apparently works fine, I have not experienced any wifi reconnection.

No, maybe a wifi disconnection or it went into hibernation... it doesn't seem to do it anymore.
I deleted the Sonoff components, I use it with Jeedom (alternative to HA), it works pretty well.

**Comment by jeroenvdd** [comment]
@seaky  maybe a suggestion for the '120 app' version include (if possible ofc) a rotate function,
since this pannel can be used in bother directions and it didn't really matter for a square one

**Comment by seaky** [comment]
> @seaky maybe a suggestion for the '120 app' version include (if possible ofc) a rotate function, since this pannel can be used in bother directions and it didn't really matter for a square one

Ok noted

**Comment by jasonanovak** [comment]
I just installed two NSPanel Pro 120s and one of them works perfectly with the proximity sensor able to wake the device. On the other one the proximity sensor seems to stop working intermittently and I get prompts that the app has stopped.

Don't know if this is something specific to 120 or something else, happy to provide logs if needed.

**Comment by Niek** [comment]
I got one as well, works flawless. Seems a bit more responsive than the regular NSPanel Pro? It also comes with a more recent webview (v107), so no need to upgrade webview.

**Comment by seaky** [comment]
> Hi,
> 
> 
> 
> I'm looking for the version of the guide for this device to make it true android :), so I can have companion app on it. Is there a version of guide describing the steps for this specific models? Based on my research, some of the steps needed for old Pro is not required for 120. Any help is appreciated. 

Szia nézd meg az nspanel roottool get out rabbit hole részt ott ki tudod nyitni az eszközt egy egyszerű otg kábellel.

**Comment by Vladx71** [comment]
So, my previous question is obsolete, it was easier than I thought. Enable developer mode, adb install launcher, adb install nspaneltools or whatever I needed. 

But, it seems waking up by proximity/wave is not working. Proximity sensors shows the correct value (1 when I about 20 cm far from the device), but the screen does not waking up. Is this some issue with nspanel tools or do I have to search for some other settings to enable this functionality?

**Comment by seaky** [comment]
On sensor page just set trigger value to 0 or 1. (I assume 0 is the right value) I dont have the device in my hand right now. That is the workaround for 120p.

Unfortunatelly you can't set trigger distance on 120p anymore with fw3.4+

New version will detect 120p automatically and will configure itself.

**Comment by seaky** [comment]
And do not forget to turn on "wake-up-wave"

**Comment by Vladx71** [comment]
Thank you. It is working differently than it is logical -to me at least :)-. When I set the threshold to 1 -for me it means proximity is detected- screen won't wake-up. If I set threshold value to 0, it works instantly.

**Comment by seaky** [comment]
for 86p it is a trigger point threshold value above the definied value the trigger will happen, that is the reason why above 0 the proximity trigger will be triggered for 120p;)

As I mentioned it is just a workaround for 120p

**Comment by Vladx71** [comment]
> > Hi,
> > I'm looking for the version of the guide for this device to make it true android :), so I can have companion app on it. Is there a version of guide describing the steps for this specific models? Based on my research, some of the steps needed for old Pro is not required for 120. Any help is appreciated.
> 
> Szia nézd meg az nspanel roottool get out rabbit hole részt ott ki tudod nyitni az eszközt egy egyszerű otg kábellel.

seems opening is totally unnecessary. adb OTA works perfectly :)

**Comment by seaky** [comment]
You mean adb command line? 
Without official sonoff app or otg method how did you enable the adb?

- if you root the panel with sonoff app you've permanently lost all the official updates
- if you "root" via OTG method it will be fully opened without loosing any future updates

**Comment by Vladx71** [comment]
Yes, with sonoff app. I was not aware I can keep future updates if I open the box... however I think it is not a big disadvantage because I suppose there will be no updates for the OS (android 8.1) and I don't really want to use ewelink app, so for me it is fine loosing future updates if I understand correctly

**Comment by seaky** [comment]
> Yes, with sonoff app. I was not aware I can keep future updates if I open the box... however I think it is not a big disadvantage because I suppose there will be no updates for the OS (android 8.1) and I don't really want to use ewelink app, so for me it is fine loosing future updates if I understand correctly

There will definitely be no relevant updates coming from the OS side, it will always remain 8.1. In fact, I don’t even recommend updating, because for example, in the case of 86p, gestures could be easily performed before version 1.10, but from 1.10 onwards, this is no longer possible. Similarly, with the 120p, the proximity sensor distance can no longer be adjusted, it’s either always 0 or 1, and this was changed at the kernel level, so I can’t modify it. The only place where an OS update might be interesting is with Zigbee, but I’ll solve that in the future to make it portable.

**Comment by seaky** [comment]
v2.2.3 intermediate version has been released

**Comment by seaky** [comment]
Just some practical comment on landscape mode. If you rotate the device first time and you use the NavigationBar in landscape mode it can appear incorrectly. Its an OS "feature". Just restart the device and will appear as expected.

Later I can add a restart popup but I assume it only needs to be set only once.

**Comment by PieterBoeren** [comment]
Thanks for this release!

Small remark, the description of the new rotation setting is wrong, it states "After 90 inactivity display sleeps".

**Comment by seaky** [comment]
> Thanks for this release!
> 
> Small remark, the description of the new rotation setting is wrong, it states "After 90 inactivity display sleeps".

Thanks for the feedback The only person who reads the description. 
The power of copy-paste has shown itself again!

I've fixed without making new version!

**Comment by seaky** [comment]
I've just got the gray panels. 
Unfortunately, it seems that Sonoff has also been affected by budget cuts or cost-cutting measures, because while the gray color of the 86P has a nice, premium metal gray look, the 120P has a rather cheap and simple gray. I would even go as far as to say it's ugly.

![ns120p](https://github.com/user-attachments/assets/09fc1430-bae0-4bab-bf31-436470e28d5d)

**Comment by jeroenvdd** [comment]
well then I was glad i was impatient and ordered all white ones because greys were on backorder, look cheapo indeed,
if i want another color, it will be done in the spraybooth :)

**Comment by JustAnotherPU** [comment]
Hi, I have just received my Nspanel Pro120 However I cannot enable adb when clicking on the device id multiple time nothing happens. Is otg the only method working ?

**Comment by seaky** [comment]
Where did you click?

**Comment by jeroenvdd** [comment]
> Hi, I have just received my Nspanel Pro120 However I cannot enable adb when clicking on the device id multiple time nothing happens. Is otg the only method working ?

wierd adb was enabled by default on all 6 of my nspro 120's

**Comment by Helvio88** [comment]
> Hi, I have just received my Nspanel Pro120 However I cannot enable adb when clicking on the device id multiple time nothing happens. Is otg the only method working ?

Same here, I clicked multiple times on Device ID, both the label and the number. Matter of fact, I multi-clicked everything in the about page. Anyone, help? NSPanel Pro 120 (Black)

**Comment by FredLiuuuu** [comment]
@seaky Thanks for the photo, I think I'll go with the white model instead of the gray one or just paint it myself ;>

**Comment by FredLiuuuu** [comment]
@Helvio88 @JustAnotherPU Try clicking in the eWeLink app, not on the screen, it should work. I was confused at first. Hope this could help.

![Image](https://github.com/user-attachments/assets/866d186e-4bf8-41f7-b2c7-54a2453a1329)
